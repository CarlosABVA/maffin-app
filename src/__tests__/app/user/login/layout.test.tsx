import React from 'react';
import { act, render } from '@testing-library/react';

import LoginLayout from '@/app/user/login/layout';

describe('LoginLayout', () => {
  it('returns loading when script not loaded', () => {
    const { container } = render(
      <LoginLayout>
        <span>child</span>
      </LoginLayout>,
    );

    expect(container).toMatchSnapshot();
  });

  it('loads renders child after loading google account script', async () => {
    const { container } = render(
      <LoginLayout>
        <span>child</span>
      </LoginLayout>,
    );

    // eslint-disable-next-line testing-library/no-node-access
    const accountsScript = document.getElementsByTagName('script')[0];
    if (accountsScript === null) {
      throw new Error('google accounts script not found in document');
    }
    expect(accountsScript.outerHTML).toEqual('<script src="https://accounts.google.com/gsi/client"></script>');

    await act(async () => {
      if (accountsScript.onload === null) {
        throw new Error('accounts script doesnt define an onload function');
      }
      await accountsScript.onload(new Event('load'));
    });

    expect(container).toMatchSnapshot();
  });
});
