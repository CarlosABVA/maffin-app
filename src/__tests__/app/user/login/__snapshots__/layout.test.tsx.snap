// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginLayout loads renders child after loading google account script 1`] = `
<div>
  <div>
    <span>
      child
    </span>
  </div>
</div>
`;

exports[`LoginLayout returns loading when script not loaded 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
